common/djangoapps/embargo/middleware.py

```diff
@@ -71,20 +84,69 @@ class EmbargoMiddleware(object):
 
     def __init__(self):
         self.site_enabled = settings.FEATURES.get('SITE_EMBARGOED', False)
+        self.enable_country_access = settings.FEATURES.get('ENABLE_COUNTRY_ACCESS', False)
+
         # If embargoing is turned off, make this middleware do nothing
-        if not settings.FEATURES.get('EMBARGO', False) and not self.site_enabled:
+        disable_middleware = not (
+            settings.FEATURES.get('EMBARGO') or
+            self.site_enabled or
+            self.enable_country_access
+        )
+        if disable_middleware:
             raise MiddlewareNotUsed()
-        self.enable_country_access = settings.FEATURES.get('ENABLE_COUNTRY_ACCESS', False)
 
     def process_request(self, request):
         """
         Processes embargo requests.
```

Comment:
I wonder if it's worth adding a docstring describing how country access, blacklists, and whitelists are used in the function and in which order.